proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -76,8 +138,16 @@ message WorkerDeploymentWorkflowArgs {
 
 // Local state for Worker Deployment
 message WorkerDeploymentLocalState {
-    string current_version = 1;
-    google.protobuf.Timestamp current_changed_time = 2;
+    google.protobuf.Timestamp create_time = 1;
+    // Encapsulates task routing information for this deployment.
+    temporal.api.deployment.v1.RoutingConfig routing_config = 2;
+    // Deployment Versions that are currently tracked in this Deployment. 
+    // A DeploymentVersion will be cleaned up automatically if all the following conditions meet:
+    // - It does not receive new executions (see WorkerDeploymentVersionInfo.accepts_new_executions) 
+    // - It has no active pollers (see WorkerDeploymentVersionInfo.pollers_status) 
+    // - It is drained (see WorkerDeploymentVersionInfo.drainage_status)
+    // TODO (Shivam) - Removal of registered versions based on the above conditions.
+    repeated string versions = 3; 
```

Comment:
Not urgent, but this can be a map from string to VersionSummary so we don't have to query all version wfs to populate version summaries when DescribeWorkerDeployment is called.